lib/ansible/modules/cloud/cloudscale/cloudscale_floating_ip.py

```diff
@@ -0,0 +1,285 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# (c) 2017, Gaudenz Steinlin <gaudenz.steinlin@cloudscale.ch>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: cloudscale_floating_ip
+short_description: Manages floating IPs on the cloudscale.ch IaaS service
+description:
+  - Create, assign and delete servers on the cloudscale.ch IaaS service.
+  - All operations are performed using the cloudscale.ch public API v1.
+  - "For details consult the full API documentation: U(https://www.cloudscale.ch/en/api/v1)."
+  - A valid API token is required for all operations. You can create as many tokens as you like using the cloudscale.ch control panel at
+    U(https://control.cloudscale.ch).
+notes:
+  - Instead of the api_token parameter the CLOUDSCALE_API_TOKEN environment variable can be used.
+  - To create a new floating IP at least the C(ip_version) and C(server) options are required.
+  - Once a floating_ip is created all parameters except C(server) are read-only.
+  - It's not possible to request a floating IP without associating it with a server at the same time.
+  - This module requires the ipaddress python library. This library is included in Python since version 3.3. It is available as a
+    module on PyPi for earlier versions.
+version_added: 2.5
+author: "Gaudenz Steinlin <gaudenz.steinlin@cloudscale.ch>"
```

Comment:
Please replace your mail with your github handle (also in cloudscale_server module)